fsspec/implementations/webhdfs.py

```diff
@@ -85,12 +85,10 @@ def __init__(self, host, port=50070, kerberos=False, token=None, user=None,
         self._connect()
 
     def _connect(self):
+        self.session = requests.Session()
         if self.kerb:
             from requests_kerberos import HTTPKerberosAuth
-            self.session = requests.Session(
-                auth=HTTPKerberosAuth(**self.kerb_kwargs))
-        else:
-            self.session = requests.Session()
+            self.session.auth = HTTPKerberosAuth(**self.kerb_kwargs)
```

Comment:
Isn't this the wrong way around - the Kerberos auth now appears to be in the `kerb is False` path. Was there a specific error with the original code?

Reply:
The original error was that `requests.Session()` does not accept `auth` as a keyword argument.

According to the change I'm adding the kerberos authentication when `self.kerb` is True.

Reply:
sorry, misread the diff colours :|